src/application/services/useMarketplace.ts

```diff
@@ -1,34 +1,43 @@
-import type EditorTool from '@/domain/entities/EditorTool';
-import { type Ref, ref, onMounted } from 'vue';
+import { type Ref, ref, watch } from 'vue';
 import { marketplaceService } from '@/domain';
+import type { EditorToolWithUserBinding } from '@/domain/entities/EditorTool';
+import { useAppState } from './useAppState';
 
 /**
  * Composable for the application state
  */
 interface UseMarketplaceComposable {
   /**
-   * All editor tools that are used in notes creation
+   * List of tools with information, if they are installed by the user
    */
-  tools: Ref<EditorTool[]>;
+  tools: Ref<EditorToolWithUserBinding[]>;
 }
 
 /**
  * Application service for working with the Editor Tools
  */
 export default function (): UseMarketplaceComposable {
+  const toolsWithUserBindings = ref<EditorToolWithUserBinding[]>([]);
+
   /**
-   *  All editor tools
+   * User tools
    */
-  const tools = ref<EditorTool[]>([]);
+  const { userEditorTools } = useAppState();
 
   /**
-   * Get list of all tools
+   * Check if user tools are changed
    */
-  onMounted(async () => {
-    tools.value = await marketplaceService.getAllTools();
-  });
+  watch(
+    userEditorTools,
+    async (newValue) => {
+      toolsWithUserBindings.value = await marketplaceService.getAllToolsWithUserBindings(newValue);
```

Comment:
you don't need to reload tools every time. Load the list (without user binding) once, then add binding in watcher. I'd suggest to move of adding `isInstalled` property to the `useMarketplace` instead of marketplace service

Reply:
replaced